### PR TITLE
[14.0] [MIG] sale_invoice_plan: Migration to 14.0

### PR DESCRIPTION
**Pull request by sonhd91**

### CODE REVIEW

File: sale_invoice_plan/models/sale.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        if invoice_plan_id:
+            plan = self.env["sale.invoice.plan"].browse(invoice_plan_id)
+            moves.ensure_one()  # Expect 1 invoice for 1 invoice plan
+            plan._compute_new_invoice_quantity(moves[0])
+            moves.invoice_date = plan.plan_date
+            plan.invoice_move_ids += moves
+        return moves
+
+
+class SaleInvoicePlan(models.Model):
```
Comment (rousseldenis) [inline sale_invoice_plan/models/sale.py]: Could you split one model per file ?

File: sale_invoice_plan/models/sale.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+            rec.last = rec.installment == last
+
+    def _compute_new_invoice_quantity(self, invoice_move):
+        self.ensure_one()
+        if self.last:  # For last install, let the system do the calc.
+            return
+        percent = self.percent
+        move = invoice_move.with_context({"check_move_validity": False})
+        for line in move.invoice_line_ids:
+            assert (
```
Comment (rousseldenis) [inline sale_invoice_plan/models/sale.py]: We are in an Odoo context, better to raise UserError instead

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

File: sale_invoice_plan/models/sale.py (hunk 1 above)
Comment (sonhd91) [reply sale_invoice_plan/models/sale.py]: Okay

File: sale_invoice_plan/models/sale.py (hunk 2 above)
Comment (sonhd91) [reply sale_invoice_plan/models/sale.py]: Okay, thanks

**Comment by sonhd91** [comment]
Hi @rousseldenis thanks for your review; I updated the PR to take into account your changes requests

**Comment by tarbib** [comment]
@rousseldenis 
Hello, looks like your requested changes were applied.
Would it be possible to update your review and merge this PR?

**Review by tarbib — APPROVED** [review]
Functional review :+1:

**Review by dsolanki-initos — APPROVED** [review]
LGTM

**Review by gurneyalex — APPROVED** [review]

**Comment by hailangvn** [comment]
runbot was failed due to SSL issue.
> WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:748)'),)': /oca-simple-and-pypi/odoo14-addon-product-supplierinfo-for-customer/
> Could not fetch URL https://wheelhouse.odoo-community.org/oca-simple-and-pypi/odoo14-addon-product-supplierinfo-for-customer/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='wheelhouse.odoo-community.org', port=443): Max retries exceeded with url: /oca-simple-and-pypi/odoo14-addon-product-supplierinfo-for-customer/ (Caused by SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:748)'),)) - skipping
> ERROR: Could not find a version that satisfies the requirement odoo14-addon-product_supplierinfo_for_customer (from odoo14-addon-product-supplierinfo-for-customer-sale) (from versions: none)
> ERROR: No matching distribution found for odoo14-addon-product_supplierinfo_for_customer

**Comment by nilshamerlinck** [comment]
Indirectly impacted by https://github.com/OCA/maintainer-quality-tools/pull/697

**Comment by nilshamerlinck** [comment]
Hi @gurneyalex the certificate issue has now been fixed; would you mind trying the merge command again?

**Pull request merged**